config/parser.go

```diff
@@ -33,14 +33,14 @@ func getStageTypeModuleName(stageType string) string {
 	return strings.ToLower(stageType)
 }
 
-func Parse(configData *map[interface{}]interface{}) (*pipelines.Pipeline, error) {
+func Parse(configData *map[interface{}]interface{}) (*pipelines.Resources, error) {
 	envs := NewEnvVariables()
 	return ParseWithSpecifiedEnvs(configData, envs)
 }
 
 // TODO: need refactoring
```

Comment:
I have not done refactoring of parser.go in this commit. I will add the following comment.

    TODO: make parser a struct (for simplifying redundant methods and reducing the number of function parameters)